### ISSUE TITLE
Viability of using Nextron alongside Elastic UI

### ISSUE THREAD

**Issue by fbraz** [saltyshiomix/nextron#29]
Sorry to bother you, I'm not sure if I should post my question here but I wanted to use nextron alongside [elastic ui](https://github.com/elastic/eui) which is a collection of React UI components.

I created a new project with `nextron init my-app --example with-javascript` and then I did `yarn add @elastic/eui @elastic/datemath moment` which installs the aforementioned library and then i did the usual stuff `yarn && yarn dev` and edited home.jsx to use eui like this:

```jsx
import React from 'react';
import Head from 'next/head';
import { resolve } from '../helpers';
import {
    EuiButton,
    EuiFlexGroup,
    EuiFlexItem,
} from '@elastic/eui';

export default () => (
  <React.Fragment>
    <Head>
      <title>Home - Nextron (with-javascript)</title>
    </Head>
    <div>
      <EuiFlexGroup gutterSize="s" alignItems="center">
          <EuiFlexItem grow={false}>
              <EuiButton
                  onClick={() => resolve('next')}>
                  Go to next pages
              </EuiButton>
          </EuiFlexItem>
      </EuiFlexGroup>
    </div>
  </React.Fragment>
);
```
I was expecting to see Nextron using EUI, but I was wrong. Instead I get a very extrange error saying `ReferenceError: HTMLElement is not defined at Object.<anonymous> (/home/ferbraz/Documents/Code/Electron/my-app/node_modules/@elastic/eui/lib/components/focus_trap/focus_trap.js:197:77) [.....]`

My question is: is it viable to use Nextron with a third party library like EUI? Do you have any idea on how to make those work together?

Sorry if I'm wrong by posting this question here, but I think that my error is related to the internals of Nextron, as EUI works with Create React App.

**Comment by saltyshiomix** [comment]
@fbraz 

Hello @fbraz !

Nextron uses Next.js in the renderer process, so it is possible to use any third party libraries which Next.js supports.

I think `@elastic/eui` does not support SSR (server side rendering).

Next.js internally runs two build processes (server and client). The `HTMLElement is not defined` error occurs when the server side webpack is processing.

**Comment by saltyshiomix** [comment]
@fbraz 

If you want to use third party libraries that do not support SSR, [electron-react-boilerplate](https://github.com/electron-react-boilerplate/electron-react-boilerplate) repo is what you are looking for.

**Comment by fbraz** [comment]
> Next.js internally runs two build processes (server and client). The `HTMLElement is not defined` error occurs when the server side webpack is processing.

Thank you @saltyshiomix for pointing me in the right direction. Now I can confirm that @elastic/eui does not support SSR, discused [here](https://github.com/elastic/eui/issues/1830). 


> If you want to use third party libraries that do not support SSR, [electron-react-boilerplate](https://github.com/electron-react-boilerplate/electron-react-boilerplate) repo is what you are looking for.

The reason why I looked at Nextron was because I was searching for a good base for electron plus react. My problem is that I really want to use an easier global state library that supports hooks, like [easy-peasy](https://github.com/ctrlplusb/easy-peasy) or [overstated](https://github.com/fabiospampinato/overstated), but I got the impression that every boilerplate is too basic or redux oriented.
I guess I will give a try to electron-react-boilerplate. I assume I could use easy-peasy and drop redux.
